lib/Db/Row2Mapper.php

```diff
@@ -599,7 +599,10 @@ public function isRowInViewPresent(int $rowId, View $view, string $userId): bool
 	 * @throws InternalError
 	 * @throws Exception
 	 */
-	public function insert(Row2 $row, array $columns): Row2 {
+	public function insert(Row2 $row, array $columns, ?string $userId = null): Row2 {
+		if ($userId) {
+			$this->userId = $userId;
+		}
```

Comment:
I know that this looks ugly. But it needed because Tables has a lot of stateful services

Reply:
Yeah, it was a total mess. There was something like permission failed error or similar.

We should use UserSession instead of injecting scalar userId. But not sure that it is a good idea to refactor it in the current PR.